generation/src/proof_generation/proofs/kore.py

```diff
@@ -355,6 +382,57 @@ def reduce_top_in_imp(self, phi: ProofThunk):
             phi,
         )
 
+    def subst_in_rewrite_target(self, phi0: Pattern, phi1: Pattern, equality: ProofThunk, imp: ProofThunk):
+        """
+        p1 k= p2        phi0  k-> next(phi1[p1/x])
+        ------------------------------------------
+                phi0  k-> next(phi1[p2/x])
+        """
+
+        # Proof Outline:
+        # 1. Axiom p1 k= p2 -> (next(phi0[p1/x])  k= next(phi0[p2/x]))
+        # 2. Axiom p1 k= p2 -> ((psi k-> p1) -> (psi k-> p2))
+        # 3. Function:
+        #    (a) p1 k= p2         (b) phi0  k-> next phi1[p1/x]
+        #    --------------------------------------------------
+        #    (c) phi0  k-> next phi1[p2/x]
+
+        # - MP on 1 and a, with subst p1 |-> p1, p2 |-> p2, phi0 |-> phi1
+        #     ==(d)=>> next(phi1[p1/x])  k= next(phi1[p2/x])
+
+        # - MP on 2 and d, with subst p1 |-> next(phi1[p1/x]), p2 |-> next(phi1[p2/x]), psi |-> phi0
+        #     ==(e)=>> ((phi0 -> (next(phi1[p1/x])) k-> (phi0 -> next(phi1[p2/x]))))
+
+        # - MP on e and b, with subst identity
+        #     ==(f)=>> phi0 -> next(phi1[p2/x]))
+
+        inner_sort, outer_sort, p1, p2 = kore_equals.assert_matches(equality.conc)
+        imp_sort, imp_left, imp_right = kore_implies.assert_matches(imp.conc) # Not needed?
```

Comment:
```suggestion
        _, _, _ = kore_implies.assert_matches(imp.conc) # Not needed?
```
Or use the corresponding method that just asserts but does not destruct

Reply:
This is actually not needed, but I kept it there while I was building the proof. It's been removed.

Reply:
Never mind. This became needed again after removing the extraneous parameters in this function. This is fixed in the latest commits.